File: js/iterators_generators/doklad/python/context_generator.py
import random

def gen_func():
	while True:
		yield random.random()

generator = gen_func()

print('generator function', gen_func)	# генераторная функция
print('generator', generator)			# генератор

print('value: ', generator.__next__())
print('value: ', generator.__next__())
print('value: ', generator.__next__())

# OUTPUT:
# value:  0.5982845385357144
# value:  0.3308434359315522
# value:  0.3165169253705432

File: js/iterators_generators/doklad/javascript/context_generator.js
function* genFunc() {
	while(true) {
		yield Math.random(1);
	}
}

console.log('generator func', genFunc);
console.log('generator', genFunc());

for (let char of genFunc()) {
  console.log('value:', char);
}


// OUTPUT:

// generator func function* genFunc() {
// 	while(true) {
// 		yield Math.random(1);
// 	}
// }
// generator Object [Generator] {}

// value: 0.1504551569680994
// value: 0.12738396889514436
// value: 0.23309275633628546
// value: 0.508185674637543
// ...
// ..
// ..
